### PR TITLE
WIP: Remove duplicate `datafusion.proto` file and use more appropriate Java package names

### PR DESCRIPTION
**Pull request by andygrove**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-datafusion/issues/2513 and https://github.com/apache/arrow-datafusion/issues/2514

 # Rationale for this change

No need to have two copies of the same source file

# What changes are included in this PR?

- Delete copy of datafusion.proto file
- Use more appropriate java package name

# Are there any user-facing changes?

Yes, for java users using these .proto files

### CODE REVIEW

File: datafusion-cli/Cargo.lock [hunk 1]
```diff
@@ -1,2531 +0,0 @@
-# This file is automatically @generated by Cargo.
```
Comment (alamb) [inline datafusion-cli/Cargo.lock:1]: I think the idea behind this file was to allow datafusion-cli to be built by itself. https://github.com/apache/arrow-datafusion/pull/2112 by @happysalada 

However, I agree it is getting annoying in my local checkout as well (seems to change regularly)

**Review by alamb — APPROVED** [review]

File: datafusion-cli/Cargo.lock (hunk 1 above)
Comment (andygrove) [reply datafusion-cli/Cargo.lock:1]: Oh, I did not mean to delete this file. I think we want to keep lock files for binaries.

**Comment by andygrove** [comment]
Failing on AMD64 with `No space left on device`

**Comment by alamb** [comment]
I think we might be able to merge this PR now

**Review by Ted-Jiang — APPROVED** [review]
Get rid of two proto files! 😊

**Pull request closed without merging**